src/sentry/workflow_engine/models/detector.py

```diff
@@ -23,15 +23,27 @@
 class Detector(DefaultFieldsModel, OwnerModel):
     __relocation_scope__ = RelocationScope.Organization
 
-    organization = FlexibleForeignKey("sentry.Organization")
+    # TODO - Finish removing this field
+    organization = FlexibleForeignKey("sentry.Organization", on_delete=models.CASCADE, null=True)
+
+    project = FlexibleForeignKey("sentry.Project", on_delete=models.CASCADE, null=True)
     name = models.CharField(max_length=200)
 
     # The data sources that the detector is watching
     data_sources = models.ManyToManyField(
         "workflow_engine.DataSource", through="workflow_engine.DataSourceDetector"
     )
 
-    # The conditions that must be met for the detector to be considered 'active'
+    # If the detector is not enabled, it will not be evaluated. This is how we "snooze" a detector
+    enabled = models.BooleanField(default=True)
+
+    # Optionally set a description of the detector, this will be used in notifications
+    description = models.TextField(blank=True, null=True)
+
+    # This represents a time delta, in seconds. If not null, this is used to determine which time
+    # window to query to compare the result from the current time_window to.
+    comparison_delta = models.IntegerField(null=True)
```

Comment:
Yeah, i agree this is a bit specific, i couldn't think of a better place for it to live at the time...

what do you think of including this on the data_source?

Reply:
I think that only makes sense if the data source is going to be performing the comparison comparation, otherwise it doesn't care. I'm not really sure what the best pattern is here, but it seems like we need somewhere to put fields that are specific to a given detector type

Reply:
chatted in slack about this; but wanted to close the loop...

it seems like there are really two solutions - creating a separate table for metric detectors and base detectors, or creating a json field with a schema. The former makes querying / data lookup a bit difficult, and we've seen a few of the downsides to that in the alerts api today.

Going to focus on making a `config` field that's a json blob. This field can have a schema that is defined on each sub-class to ensure consistence there, and enforce the type checking on the base `detector` model.